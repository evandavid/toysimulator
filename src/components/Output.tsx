import React from 'react';
import {View} from 'react-native';
import {ROBOT_POSITION} from '../util';
import {CommandText} from './styled';

const OutputView = ({data}: {data: ROBOT_POSITION[]}) => {
  return (
    <View>
      {data.map((datum, index) => (
        <View key={'command' + index}>
          <CommandText>
            {'>>'} {`Y:${datum.row}, X:${datum.col}, ${datum.f}`}
          </CommandText>
        </View>
      ))}
    </View>
  );
};

export default OutputView;
