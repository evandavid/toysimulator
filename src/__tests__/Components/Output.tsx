import React from 'react';
import renderer from 'react-test-renderer';
import Output from '../../components/Output';

it('renders correctly', () => {
  const tree = renderer
    .create(<Output data={[{col: 0, row: 0, f: 'EAST'}]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
