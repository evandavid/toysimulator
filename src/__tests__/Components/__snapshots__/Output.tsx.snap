// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View>
    <Text
      className="sc-hKgJUU gfmGt"
    >
      &gt;&gt;
       
      Y:0, X:0, EAST
    </Text>
  </View>
</View>
`;
